PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -317,8 +317,16 @@ void ScintillaEditView::init(HINSTANCE hInst, HWND hPere)
 		isWINE = ::GetProcAddress(hNtdllModule, "wine_get_version");
 
 	if (isWINE || // There is a performance issue under WINE when DirectWrite is ON, so we turn it off if user uses Notepad++ under WINE
-		::IsWindowsServer()) // In the case of Windows Server Core, DirectWrite cannot be on.
-		nppGui._writeTechnologyEngine = defaultTechnology;
+		isCoreWindows()) // In the case of Windows Server Core, DirectWrite cannot be on.
+	{
+		nppGui._writeTechnologyEngine = directWriteTechnologyUnavailable;
+	}
+	else
+	{
+		// allow IDC_CHECK_DIRECTWRITE_ENABLE to be set in Preferences > MISC. again
+		if (nppGui._writeTechnologyEngine == directWriteTechnologyUnavailable)
+			nppGui._writeTechnologyEngine = defaultTechnology;

```

Comment:
👍

Reply:
Just re-read the code, here nppGui._writeTechnologyEngine will never be directWriteTechnologyUnavailable, because of this line: `_nppGUI._writeTechnologyEngine = (lstrcmp(optNameWriteTechnologyEngine, L"1") == 0) ? directWriteTechnology : defaultTechnology;` in https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/Parameters.cpp#L6189

Nevertheless, we can keep it - it shows more clearly our intention.

Reply:
@donho 
Ok, you are right, I missed that.

I only saw that in the config.xml that `writeTechnologyEngine="2"` has been really written there and I imagined that someone might have a portable N++ setup on a usb-flash and once used it in Wine or Core env, that checkbox could have been disabled forever.